src/test/java/mate/academy/security/jwt/JwtTokenProviderTest.java

```diff
@@ -0,0 +1,183 @@
+package mate.academy.security.jwt;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.any;
+import static org.springframework.security.core.userdetails.User.UserBuilder;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import mate.academy.exception.InvalidJwtAuthenticationException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.test.util.ReflectionTestUtils;
+
+class JwtTokenProviderTest {
+    private static final String SECRET_KEY = "secret";
+    private static final long VALIDITY_IN_MILLISECONDS = 3600000L;
+    private UserDetailsService userDetailsServiceMock;
+    private JwtTokenProvider jwtTokenProvider;
+    private String email;
+    private List<String> roles;
+
+    @BeforeEach
+    void setUp() {
+        userDetailsServiceMock = Mockito.mock(UserDetailsService.class);
+        jwtTokenProvider = new JwtTokenProvider(userDetailsServiceMock);
+        ReflectionTestUtils.setField(jwtTokenProvider, "secretKey", SECRET_KEY);
+        ReflectionTestUtils.setField(jwtTokenProvider, "validityInMilliseconds",
+                VALIDITY_IN_MILLISECONDS);
+        email = "user@gmail.com";
+        roles = List.of("ADMIN");

```

Comment:
I cannot see any change of these variable in tests, why to set it before each one?